precompiles/staking/IStaking.sol

```diff
@@ -26,6 +26,38 @@ struct Validator {
 }
 
 interface IStaking {
+    /// @notice Stake event is emitted when stake function is called
+    /// @param staker Staker address
+    /// @param validator Validator address
+    /// @param amount Coins amount
+    event Stake(
+        address indexed staker,
+        address indexed validator,
+        uint256 amount
+    );
```

Comment:
Does the calling of msgServer.Delegate , 
automatically create the cosmos event ? 

```
	ctx.EventManager().EmitEvents(sdk.Events{
		sdk.NewEvent(
			types.EventTypeDelegate,
			sdk.NewAttribute(types.AttributeKeyValidator, msg.ValidatorAddress),
			sdk.NewAttribute(types.AttributeKeyDelegator, msg.DelegatorAddress),
			sdk.NewAttribute(sdk.AttributeKeyAmount, msg.Amount.String()),
			sdk.NewAttribute(types.AttributeKeyNewShares, newShares.String()),
		),
	})

```

If so why do we need this event as well ? It seems to have duplicate information